### PR TITLE
Adding guide for product teams + core instrumenting spans

### PR DESCRIPTION
**Pull request by patrickhuie19**

### CODE REVIEW

File: .github/tracing/README.md [hunk 1]
```diff
@@ -48,4 +48,49 @@ This folder contains the following config files:
...
\ No newline at end of file
+`otel-collector-ci.yaml` is the configuration for the CI runs, and exports the trace data to the artifact from the github run.
+
+## Instrumenting Custom Traces
+
+Adding custom traces requires identifying an observability gap in a related group of code executions. This is intuitive for the developer:
+
+- "What's the flow of component interaction in this distributed system?"
+- "What's a ballpark latency of funcOne?"
+- "What's the behavior of the JobProcessorOne component when jobs with [x, y, z] attributes are processed?"
```
Comment (krehermann) [inline .github/tracing/README.md:59]: too abstract. i don't understand what you are trying to express. suggest remove or make concrete

File: .github/tracing/README.md [hunk 2]
```diff
@@ -48,4 +48,49 @@ This folder contains the following config files:
...
+
+Once an execution flow is desired to be traced, the developer will decide which parts of the execution flow should be measured separately. Each logically separate measure will be a span. Spans can have one parent span and multiple children span. The relations between parent and child spans will make a directed acyclic graph.
```
Comment (krehermann) [inline .github/tracing/README.md:61]: `Once an execution flow is desired to be traced, the developer will decide which parts of the execution flow should be measured separately` reads poorly. how about 'given an execution flow, the developer must decide which subset of the flow to measure'

`will make a directed acyclic graph` with a single root?

File: .github/tracing/README.md [hunk 3]
```diff
@@ -48,4 +48,49 @@ This folder contains the following config files:
...
+
+The most trivial application of a span is measuring latency performance at high levels of granularity (individual func executions). There is much more you can do, including creating human readable and timestamped events within a span (useful for monitoring concurrent access to resources), recording errors, linking parent and children spans through large parts of an application, and even extending a span beyond a single process.
```
Comment (krehermann) [inline .github/tracing/README.md:63]: are there otel best practices we can link to? you don't wan the reader to interpret this as a suggestion to always trace every func. that's a lot of manual work. presumably there is an 80/20 rule wrt to return on investment of manual instrumentation. maybe otel or some other reputable source has guidelines

Comment (patrickhuie19) [reply .github/tracing/README.md:63]: I searched for a primary-source from OTEL on best practices but didn't find anything. This is a good 3rd party source: https://www.zerok.ai/post/distributed-tracing-best-practices

File: .github/tracing/README.md (hunk 2 above)
Comment (patrickhuie19) [reply .github/tracing/README.md]: Yes, each trace will have a single root.

**Review by krehermann — APPROVED** [review]

**Comment by patrickhuie19** [comment]
Going to merge for now; comments always welcome

**Pull request merged**